wagtail/images/templatetags/wagtailimages_tags.py

```diff
@@ -154,3 +155,200 @@ def image_url(image, filter_spec, viewname="wagtailimages_serve"):
+            preserve_svg = True
+        else:
+            try:
+                name, value = bit.split("=")
+                attrs[name] = parser.compile_filter(
+                    value
+                )  # setup to resolve context variables as value
+            except ValueError:
+                if allowed_filter_pattern.match(bit):
+                    filter_specs.append(bit)
+                else:
+                    raise template.TemplateSyntaxError(
+                        "filter specs in 'image' tag may only contain A-Z, a-z, 0-9, dots, hyphens and underscores. "
+                        "(given filter: {})".format(bit)
+                    )
+
+    if as_context and output_var_name is None:
+        # context was introduced but no variable given ...
+        is_valid = False
+
+    if output_var_name and attrs:
+        # attributes are not valid when using the 'as img' form of the tag
+        is_valid = False
+
+    if len(filter_specs) == 0:
+        # there must always be at least one filter spec provided
+        is_valid = False
+
+    if len(bits) == 0:
+        # no resize rule provided eg. {% image page.image %}
+        raise template.TemplateSyntaxError(
+            "no resize rule provided. "
+            "'image' tag should be of the form {% image self.photo max-320x200 [ custom-attr=\"value\" ... ] %} "
+            "or {% image self.photo max-320x200 as img %}"
+        )
+
+    if is_valid:
+        return ImageSrcsetNode(
+            image_expr,
+            filter_specs,
+            attrs=attrs,
+            output_var_name=output_var_name,
+            preserve_svg=preserve_svg,
+        )
+    else:
+        raise template.TemplateSyntaxError(
+            "'image' tag should be of the form {% image self.photo max-320x200 [ custom-attr=\"value\" ... ] %} "
+            "or {% image self.photo max-320x200 as img %}"
+        )
+
+class ImageSrcsetNode(template.Node):
+    def __init__(
+        self,
+        image_expr,
+        filter_specs,
+        output_var_name=None,
+        attrs={},
+        preserve_svg=False,
+    ):
+        self.image_expr = image_expr
+        self.output_var_name = output_var_name
+        self.attrs = attrs
+        self.filter_specs = filter_specs
+        self.preserve_svg = preserve_svg
+
+    @lru_cache()
+    def raw_filter_specs(self, context):
+        # Space-separated filters, to be concatenated.
+        spaced_filters = []
+        # Brace-expanded filters, to generate multiple renditions at once.
+        braced_filters = []
+
+        named_filters = getattr(settings, 'WAGTAIL_PICTURE_PROPOSAL_NAMED_FILTERS', {})
+
+        for spec in self.filter_specs:
+            raw_filter = spec.resolve(context) if isinstance(spec, FilterExpression) else spec
+            # TODO If the filter matches one of the predefined named filters.
+            # Do we want those to be expanded as well?
+            raw_filter = named_filters.get(raw_filter, raw_filter)
```

Comment:
I’d suggest removing this for now for the sake of simplicity. Can always be added later.